### PR TITLE
SDK-183 Java BDK: Create test coverage for bot.sdk.command.MultiResponseComposerImpl

### PR DESCRIPTION
**Pull request by Haiddi-Avencod**
### Ticket
[SDK-183](https://perzoinc.atlassian.net/browse/SDK-183?atlOrigin=eyJpIjoiZWM3MGI5MzZkMzRjNDIwM2IzZjBlNjRjNTE4MDJkNzciLCJwIjoiaiJ9)

### Coverage
coverage_183

### CODE REVIEW

File: symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/command/MultiResponseComposerImplTest.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    assertEquals(expectedComposedResponse, this.multiResponseComposerImpl.getComposedResponse());
+  }
+
+  @Test
+  public void testWithMessage() {
+    final String message = "message1";
+
+    final ComposerAttachmentOrStreamDefinition composerAttachmentOrStreamDefinition =
+        this.multiResponseComposerImpl.withMessage(message);
+    assertEquals(composerAttachmentOrStreamDefinition, this.multiResponseComposerImpl);
```
Comment (symphony-elias) [inline symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/command/MultiResponseComposerImplTest.java:53]: To check that message has been added into `this.multiResponseComposerImpl`, please call `this.multiResponseComposerImpl.toStreams("streamId")` and `this.multiResponseComposerImpl.getComposedResponse()` to check if the map is the expected one.
Please do the same for other methods `with*`.

Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/command/MultiResponseComposerImplTest.java:53]: I tried that : 

```
  @Test
  public void testWithMessage() {
    final String message = "message1";

    final ComposerAttachmentOrStreamDefinition composerAttachmentOrStreamDefinition =
        this.multiResponseComposerImpl.withMessage(message);
    assertEquals(composerAttachmentOrStreamDefinition, this.multiResponseComposerImpl);

    this.multiResponseComposerImpl.toStreams(message);
    final Map<SymphonyMessage, Set<String>> composedResponse = this.multiResponseComposerImpl.getComposedResponse();
    assertEquals(message, composedResponse.get(message));
  }
```

but it doesn't work. Could you explain me how to do please ?

Comment (Haiddi-Avencod) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/command/MultiResponseComposerImplTest.java:53]: Like that :
```
  @Test
  public void testWithMessage() {
    final String message = "message1";

    final ComposerAttachmentOrStreamDefinition composerAttachmentOrStreamDefinition =
        this.multiResponseComposerImpl.withMessage(message);
    assertEquals(composerAttachmentOrStreamDefinition, this.multiResponseComposerImpl);

    this.multiResponseComposerImpl.toStreams("streamId");
    final Map<SymphonyMessage, Set<String>> composedResponse = this.multiResponseComposerImpl.getComposedResponse();
    assertEquals(message, composedResponse.get("streamId"));
  }
```
  
  it doesn't work either

Comment (symphony-elias) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/command/MultiResponseComposerImplTest.java:53]: I realized you have to call the `compose()` method first. This is working:
```java
@Test
  public void testComposeWithMessage() {
    final String message = "message1";
    final String streamId = "streamId";

    this.multiResponseComposerImpl.compose().withMessage(message);
    this.multiResponseComposerImpl.toStreams(streamId);

    final Map<SymphonyMessage, Set<String>> composedResponse = this.multiResponseComposerImpl.getComposedResponse();
    final List<Map.Entry<SymphonyMessage, Set<String>>> entries = new ArrayList(composedResponse.entrySet());

    assertEquals(1, entries.size());

    final Map.Entry<SymphonyMessage, Set<String>> entry = entries.get(0);
    assertEquals(message, entry.getKey().getMessage());
    assertEquals(Collections.singleton(streamId), entry.getValue());
  }
```

**Review by symphony-elias — APPROVED** [review]
Looks good !

**Pull request merged**